File: sidecars/rex-agent/src/rex_agent/advisory_intent.py
"""Advisory prompt intent detection (R067 / R070)."""

from __future__ import annotations

ADVISORY_PATTERNS: tuple[str, ...] = (
    "what should we do next",
    "what's next",
    "what to work on",
    "priorities",
    "roadmap",
    "next step",
)

PRIORITY_CONTEXT_MARKERS: tuple[str, ...] = (
    "roadmap",
    "prioritization",
    "current focus",
)


def matches_advisory_intent(text: str) -> bool:
    lower = (text or "").lower()
    return any(pattern in lower for pattern in ADVISORY_PATTERNS)


def daemon_context_has_priority_markers(text: str) -> bool:
    lower = (text or "").lower()
    return any(marker in lower for marker in PRIORITY_CONTEXT_MARKERS)

File: crates/rex-daemon/src/advisory_intent.rs
//! Advisory prompt intent detection (R067 / R070).

const ADVISORY_PATTERNS: &[&str] = &[
    "what should we do next",
    "what's next",
    "what to work on",
    "priorities",
    "roadmap",
    "next step",
];

/// True when the prompt asks for priorities, roadmap, or next-work guidance.
pub fn matches_advisory_intent(prompt: &str) -> bool {
    let lower = prompt.to_ascii_lowercase();
    ADVISORY_PATTERNS
        .iter()
        .any(|pattern| lower.contains(pattern))
}

/// True when injected daemon context already carries roadmap/priority signals.
/// Retained for R070 sidecar parity; daemon init gating lives in the sidecar.
#[allow(dead_code)]
pub fn context_has_priority_markers(context: &str) -> bool {
    let lower = context.to_ascii_lowercase();
    lower.contains("roadmap") || lower.contains("prioritization") || lower.contains("current focus")
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn matches_short_advisory_prompt() {
        assert!(matches_advisory_intent("What should we do next?"));
    }

    #[test]
    fn skips_unrelated_short_prompt() {
        assert!(!matches_advisory_intent("hello"));
    }
}
